### PR TITLE
GCP: improve integration docs

### LINKED ISSUE #2802: Improve Google Cloud Platform (GCP) Agent integration doc
## Current problem
The current Elastic Agent GCP integration doesn’t provide any details to users on how to configure the integration.
Here’s what we say in the integrations UI and the integration doc:
Screen Shot 2022-03-09 at 11 40 17 AM
Screen Shot 2022-03-09 at 11 39 19 AM
We do not provide any details on how to configure the GCP integration and the user is pretty much on his own in this screen.
Screen Shot 2022-03-09 at 11 58 19 AM

### Authentication:
Enter the **Project Id** and either the **_Credentials File_** or **_Credentials Json_** value for Elastic agent to authenticate with the Google cloud platform and be able to collect observability data.

You need to create a Service Account and one Service Account Key, which are the credentials associated with the Service Account. To monitor GCP services (logs and metrics) you need following roles assigned to the service account:

- Compute Viewer
- Monitoring Viewer
- Pub/Sub Subscriber

Download the service account private key as a JSON file to be used by the Elastic Agent for authentication. Specify the information in either the Credentials File OR Credentials Json field based on your preference.

**Note:** Elastic recommends using **_Credentials File_** as in this method the credential information doesn’t leave your Google cloud platform environment. When using **_Credentials Json_** the information is stored in Elasticsearch and the access is controlled based on policy permissions or access to underlying Elasticsearch data.

**Credentials File:** Save the credentials file in a secure location and make sure that Elastic agent has at least read only privileges to this file. Specify the file path in the **_Credentials File_** field in the Elastic Agent GCP integration UI.
**Example:** "/home/ubuntu/credentials.json"

**Credentials Json:** Specify the Json content directly in the **_Credentials Json_** field in the Elastic Agent GCP integration

The **_Project Id_** and either the **_Credentials File_** or **_Credentials Json_** will need to be provided in the integration UI when adding the Google Cloud Platform (GCP) integration.

Please review following in Google Cloud Platform documentation for more details:
[Creating and managing projects](https://cloud.google.com/resource-manager/docs/creating-managing-projects)
[Authentication overview](https://cloud.google.com/docs/authentication)
[Authenticating as a service account](https://cloud.google.com/docs/authentication/production)
[Choose predefined roles](https://cloud.google.com/iam/docs/choose-predefined-roles)

## Log collection configuration:
Enter Google Pub/Sub **_Topic_** and **_Subscription name_** for each of the log types (audit, VPC flow, DNS and firewall logs) you want to collect. You may need to create log Sinks and Pub/Sub Topics based on your needs. Elastic recommends separate Pub/Sub topics for each of the log types so that they can be parsed and stored in a specific data stream.

Elastic Agent periodically fetches Google Cloud logs that have been exported from Cloud Logging to a Google Pub/Sub topic sink.  Here’s an example on how to create a Pub/Sub topic and subscription using **_Log Router_** in the Google cloud console. At a high level the steps required are:

- **_Create sink_** and provide a sink name and description. 
- Select **_Cloud Pub/Sub topic_** as the sink service. Select an existing topic or **_Create a topic_**. Note the topic name, as it will be provided in the **_Topic_** field in the Elastic agent configuration. If you created a new topic then you must remember to go to that topic and create a subscription for it. A subscription directs messages on a topic to subscribers. Note the Subscription ID as it will need to be entered in the **_Subscription name_** field in the Elastic agent configuration.
- Under **_Choose logs to include in sink_**, for example add `logName:"cloudaudit.googleapis.com"` in the **_Inclusion filter_** to include all audit logs. This is just an example, you will need to create your own filter expression to select the log types you want to export to the Pub/Sub topic.

The **_Topic_** and **_Subscription name_** will need to be provided in the integrations UI when adding the Google cloud platform (GCP) integration.

Please review following in Google cloud documentation for more details:
[Routing and storage overview](https://cloud.google.com/logging/docs/routing/overview)
[Create a topic](https://cloud.google.com/pubsub/docs/create-topic-console#create_a_topic)
[Configure and manage sinks](https://cloud.google.com/logging/docs/export/configure_export_v2)
[Logging query language](https://cloud.google.com/logging/docs/view/logging-query-language)

### PR DESCRIPTION
**Pull request by zmoog**
## What does this PR do?

Update README for the Google Cloud Platform (GCP) integration with new sections about how to configure authentication, integration settings, and how to start collecting logs creating a pub/sub topic, a subscription, and a sink.

The fields and events sample are included in both the README and the dedicated `.md` files, in preparation of making them visible on the main docs site https://docs.elastic.co/en/integrations/gcp.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

Closes #2802

### CODE REVIEW

File: packages/gcp/_dev/build/docs/README.md [hunk 1]
```diff
@@ -1,38 +1,192 @@
...
+
+Specify the file path in the Elastic Agent integration UI in the "Credentials File" field. For example: `/home/ubuntu/credentials.json`.
+
+#### Option 2: Credentials Json
+
+Specify the content of the JSON file you downloaded from Google Cloud Platform directly in the Credentials Json field in the Elastic Agent integration.
+
+#### Recommendations
+
+Elastic recommends using Credentials File as this method, so the credential information doesn’t leave your Google Cloud Platform environment. When using Credentials Json, the integration stores the info in Elasticsearch, and the access is controlled based on policy permissions or access to underlying Elasticsearch data.
```
Comment (ravikesarwani) [inline packages/gcp/_dev/build/docs/README.md:68]: Maybe reword to "Elastic recommends using Credentials File, as in this method the credential...."

File: packages/gcp/_dev/build/docs/README.md [hunk 2]
```diff
@@ -1,38 +1,192 @@
...
+At a high level, the steps required are:
+
+- Visit "Logging" > "Log Router" > "Create Sink" and provide a sink name and description.
+- In "Sink destination", select "Cloud Pub/Sub topic" as the sink service. Select an existing topic or "Create a topic". Note the topic name, as it will be provided in the Topic field in the Elastic agent configuration.
+- If you created a new topic, you must remember to go to that topic and create a subscription for it. A subscription directs messages on a topic to subscribers. Note the "Subscription ID", as it will need to be entered in the "Subscription name" field in the ingtegration settings.
+- Under "Choose logs to include in sink", for example add `logName:"cloudaudit.googleapis.com"` in the "Inclusion filter" to include all audit logs.
+
+This is just an example; you will need to create your filter expression to select the log types you want to export to the Pub/Sub topic.
+
+More example filters for different log types:
```
Comment (ravikesarwani) [inline packages/gcp/_dev/build/docs/README.md:97]: From just the look of it the sample seems pretty inconsistent. If we are providing all the samples then I would say let's make sure at some level that it works.

File: packages/gcp/_dev/build/docs/README.md [hunk 3]
```diff
@@ -1,38 +1,192 @@
...
+#
+resource.type="dns_query"
+
+#
+# All Firewall logs
+#
+logName="projects/PROJECT_ID/logs/compute.googleapis.com%2Ffirewall"
+```
+
+### On the Google Cloud Integration Settings
```
Comment (ravikesarwani) [inline packages/gcp/_dev/build/docs/README.md:121]: I think this transition from Google console to Kibana is a little difficult to understand. Few suggestion:
- Make sure we mention that this is in Kibana
- Not sure why the heading says "On the Google Cloud Integration Settings". What are we changing in the settings? Feels like we are configuring the integration.

File: packages/gcp/_dev/build/docs/README.md [hunk 4]
```diff
@@ -1,38 +1,192 @@
 # Google Cloud Integration
 
-The Google Cloud integration collects and parses Google Cloud audit, VPC flow,
-and firewall logs that have been exported from Stackdriver to a Google Pub/Sub topic sink.
+## Overview
+
+The Google Cloud integration collects and parses Google Cloud [Audit Logs](https://cloud.google.com/logging/docs/audit), [VPC Flow Logs](https://cloud.google.com/vpc/docs/using-flow-logs), [Firewall Rules Logs](https://cloud.google.com/vpc/docs/firewall-rules-logging) and [Cloud DNS Logs](https://cloud.google.com/dns/docs/monitoring) that have been exported from Stackdriver to a Google Pub/Sub topic sink.
+
```
Comment (ravikesarwani) [inline packages/gcp/_dev/build/docs/README.md:6]: Stackdriver is old name. We should change it to Cloud Logging.

Comment (zmoog) [reply packages/gcp/_dev/build/docs/README.md:6]: Thanks, I copied from an existing doc without double checking id this term was still the best possible. Amended!

File: packages/gcp/_dev/build/docs/README.md [hunk 5]
```diff
@@ -1,38 +1,192 @@
...
+Solution: double check the integration settings.
 
 ## Logs
 
 ### Audit
 
-This is the `audit` dataset.
-
-{{event "audit"}}
+The `audit` dataset collects audit logs of administrative activities and accesses within your Google Cloud resources.
```
[the remark this answers is not in the document]
Comment (zmoog) [reply packages/gcp/_dev/build/docs/README.md:164]: Yeah, I wrote down a note for myself to get rid of the events and fields section in README once we're ready to ship the docs in multiple files.

File: packages/gcp/_dev/build/docs/README.md (hunk 3 above)
Comment (zmoog) [reply packages/gcp/_dev/build/docs/README.md:121]: You're right, and I can see your point here. The title somewhat made sense in my mental model of this thing, but I now see it's not right, so I'm amending it.

File: packages/gcp/_dev/build/docs/README.md [hunk 6]
```diff
@@ -1,38 +1,192 @@
...
+
+The "Project Id" and either the "Credentials File" or "Credentials Json" will need to be provided in the integration UI when adding the Google Cloud Platform integration.
+
+### Project Id
+
+The Project Id is the Google Cloud project ID where your resources exist.
+
+### Credentials File vs Json
+
+Based on your preference, specify the information in either the Credentials File OR the Credentials Json field.
```
Comment (efd6) [inline packages/gcp/_dev/build/docs/README.md:54]: nit: Should the integration be fixed to have correct capitalisation on this (Credentials **Json**) field?

File: packages/gcp/_dev/build/docs/README.md [hunk 7]
```diff
@@ -1,38 +1,192 @@
...
 {{fields "firewall"}}
 
-### VPC Flow
+{{event "firewall"}}
 
-This is the `VPC Flow` dataset.
+### VPC Flow
 
-{{event "vpcflow"}}
+he `vpcflow` dataset collects logs sent from and received by VM instances, including instances used as GKE nodes.
```
Comment (efd6) [inline packages/gcp/_dev/build/docs/README.md:180]: "The"

File: packages/gcp/_dev/build/docs/README.md (hunk 6 above)
Comment (zmoog) [reply packages/gcp/_dev/build/docs/README.md:54]: Yeah, I was thinking the same, the current capitalisation keeps bugging me every time I read it: @ravikesarwani do you think we can rename it from "Credentials Json" to "Credentials JSON"?

File: packages/gcp/_dev/build/docs/README.md (hunk 7 above)
Comment (zmoog) [reply packages/gcp/_dev/build/docs/README.md:180]: Oh God, thank you for picking it up 🙇

File: packages/gcp/_dev/build/docs/README.md (hunk 2 above)
Comment (zmoog) [reply packages/gcp/_dev/build/docs/README.md:97]: Reworked the samples making them more consistent. 

Also added a link to the Logs Explorer, relevant Google-provided docs and samples.

Comment (ravikesarwani) [reply packages/gcp/_dev/build/docs/README.md:97]: Maurizio: For Audit logs both [logName:"cloudaudit.googleapis.com"] and [logName:(projects/PROJECT_ID/logs/cloudaudit.googleapis.com)] will work as a filter when setting up Log Router? Did we actually tried it?

VPC flow logs sample uses "=" [
logName="projects/PROJECT_ID/logs/compute.googleapis.com%2Fvpc_flows"] but Cloud Audit logs uses ":"[logName:(projects/PROJECT_ID/logs/cloudaudit.googleapis.com)] and seems a little inconsistent.
Do we know they both work by actually trying it out?

I have tried only [logName:"cloudaudit.googleapis.com"].
My take would be in the first iteration lets add only the samples that we have tried and made to work in a real environment by testing it out. For anything else let's link to the Google-provided docs and samples.

Comment (ravikesarwani) [reply packages/gcp/_dev/build/docs/README.md:97]: Looks good.

**Review by ravikesarwani — APPROVED** [review]
Good from my side. Great Work!

**Comment by zmoog** [comment]
@efd6 do you think we're good to go or this PR needs some changes from the [elastic/security-external-integrations](https://github.com/orgs/elastic/teams/security-external-integrations) perspective?

**Comment by ravikesarwani** [comment]
@zmoog My suggestion would be to get this merged and promoted to production. My take would be any additional changes can be taken up as incremental work based on users feedback with a separate issue and PR. Thanks!

**Comment by zmoog** [comment]
@ravikesarwani sounds good, let's make it so.

**Comment by ravikesarwani** [comment]
I know there's also a Kibana dependency that we have to specify. Since this is basically a doc and minor changes I would like to see us support as older version of 7.x **as possible** along with 8.0+. This way more customers can see this and benefit by it.

**Comment by zmoog** [comment]
@ravikesarwani the current Kibana dependency constraint is set as follow:

```yaml
conditions:
  kibana.version: ^7.16.3 || ^8.0.0
```

It was set by Andrew Kroh a few months ago: do you think we need to change it?

**Comment by ravikesarwani** [comment]
No, this fine.
